library/core/src/macros/mod.rs

```diff
@@ -17,6 +17,42 @@ macro_rules! panic {
     );
 }
 
+#[unstable(feature = "panic_internals", reason = "interal API", issue = "none")]
+#[macro_export]
+#[doc(hidden)]
+#[allow_internal_unstable(core_panic, panic_internals)]
+macro_rules! assert_binary_cmp {
+    ($left:expr, $op:tt, $right:expr, $msg:expr) => ({
+        match (&$left, &$right) {
+            (left_val, right_val) => {
+                if !(*left_val $op *right_val) {
+                    let assertion = $crate::panic::assert_info::BinaryAssertion {
+                        op: stringify!($op),
+                        left_expr: stringify!($left),
+                        right_expr: stringify!($right),
+                        left_val: &&*left_val,
+                        right_val: &&*right_val,
+                    };
+
+                    // The reborrows below are intentional. Without them, the stack slot for the
+                    // borrow is initialized even before the values are compared, leading to a
+                    // noticeable slow down.
+                    $crate::panicking::start_panic(
+                        $crate::option::Option::None,
+                        &$crate::format_args!(r#"assertion failed: `(left {} right)`
+  left: `{:?}`,
+ right: `{:?}`"#, stringify!($op), &*left_val, &*right_val),
```

Comment:
This is missing the `: {}` when a message is given.

---

(Currently, `assert_eq!(1, 2, "hello")` panics with the message:

```
assertion failed: `(left == right)`
  left: `1`,
 right: `2`: hello
```
But this changed version is missing the `: hello`.)